### PR TITLE
add structured logging & better reservation values

### PR DESCRIPTION
**Pull request by natebrennand**
ticket: https://clever.atlassian.net/browse/DOC-2069

Match https://github.com/Clever/node-redis-reservation/pull/14

### CODE REVIEW

File: reservation/reservation.go [hunk 1]
```diff
@@ -47,20 +51,18 @@ func NewManager(redisURL, owner string) (*Manager, error) {
...
-	// Get hostname
-	hostname, err := os.Hostname()
-	if err != nil {
-		return nil, err
-	}
-
-	key := fmt.Sprintf("reservation-%s", resource)
-	val := fmt.Sprintf("%s-%s-%d", hostname, manager.owner, os.Getpid())
+	key := redisKey(resource)
+	val := fmt.Sprintf("%s-%s", manager.owner, os.Getenv("JOB_ID"))
```
Comment (peggyl) [inline reservation/reservation.go]: 👏

File: reservation/reservation.go [hunk 2]
```diff
@@ -74,9 +76,11 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 		"EX", manager.TTL.Seconds(),
 		"NX")
 	if err != nil {
+		manager.lg.ErrorD("redis-error", logger.M{"key": key, "resource": resource, "err": err.Error()})
 		return nil, fmt.Errorf("Error with SET command: %s", err.Error())
 	}
 	if success == nil {
+		manager.lg.InfoD("reservation-exists", logger.M{"key": key, "resource": resource})
```
Comment (peggyl) [inline reservation/reservation.go]: Should this be ErrorD if you're returning an error?

**Comment by natebrennand** [comment]
> I didn't know source files had to be executable to import or test… 😭

That's a commit message derp... I was removing their right to execute

File: reservation/reservation.go (hunk 2 above)
Comment (natebrennand) [inline reservation/reservation.go]: `InfoD` mostly to match the node implementation: https://github.com/Clever/node-redis-reservation/blob/95eb69186b7c31ed60689123b45e7d1dca717e7c/lib/index.coffee#L41

I'm not in love with this API but it isn't worth switching it up either IMO.

**Pull request merged**